spec/datadog/tracing/contrib/karafka/monitor_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'datadog/tracing/contrib/support/spec_helper'
+require 'datadog/tracing/contrib/analytics_examples'
+
+require 'karafka'
+require 'datadog'
+
+RSpec.describe 'Karafka monitor' do
+  subject(:monitor) { described_class.new }
+  let(:configuration_options) { { distributed_tracing: true } }
+
+  before do
+    Datadog.configure do |c|
+      c.tracing.instrument :karafka, configuration_options
+    end
+  end
+
+  around do |example|
+    # Reset before and after each example; don't allow global state to linger.
+    Datadog.registry[:karafka].reset_configuration!
+    example.run
+    Datadog.registry[:karafka].reset_configuration!
+  end
+
+  describe '.instrument' do
+    context 'when the event is not traceable' do
+      it 'does not create a trace' do
+        allow(Datadog::Tracing).to receive(:trace)
+
+        Karafka.monitor.instrument('worker.completed')
+
+        expect(Datadog::Tracing).not_to have_received(:trace)
+      end
+    end
+
+    context 'when the event is traceable' do
+      let(:span) { instance_double('Span', set_tag: nil) }
```

Comment:
There is a real class behind Span and instance_double will not help here to avoid stubbing.

It's `Datadog::Tracing::SpanOperation`. But I would recommend to follow suggestion below to avoid a tight coupling with implementation.